include/ts/sdt.h

```diff
@@ -0,0 +1,46 @@
+/** @file
+
+    A brief file description
+
+    @section license License
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#ifdef HAVE_SYSTEMTAP
+
+#include <sys/sdt.h>
+
+#else
+
+#define DTRACE_PROBE(...)
```

Comment:
I should have commented a bit earlier, but having these might be problematic in terms of license. Maybe I'm worrying too much, but it may seem like we stole the interface.

Reply:
I think we have three options.

1) Remove null DTRACE_*  macros and have HAVE_SYSTEMTAP here and there
This would be a safe bet, however having many ifdef on main path would be annoying.

2) Keep this as is and have an attribution on NOTICE file
Like I said on my comment above, having the null macros might be problematic because we copied the interface. As long as the license is compatible we can do it, however we need to add an attribution to NOTICE file.
https://www.apache.org/legal/resolved.html#handling-public-domain-licensed-works

3) Define our own interface and use DTRACE_* as an implementation if it's available
I think this is an ideal option. No ifdef on main path, and no license issue.


I'd say take 1) for now and 3) in the future.

I'm also ok with 2), however, we might need to maintain the interface if the original interface was changed (otherwise copying interface doesn't make sense).